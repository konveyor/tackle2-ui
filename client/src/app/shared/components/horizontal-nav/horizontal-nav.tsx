import React from "react";
import { NavLink } from "react-router-dom";

export interface HorizontalNavProps {
  navItems: { title: string; path: string }[];
}

export const HorizontalNav: React.FC<HorizontalNavProps> = ({ navItems }) => {
  return (
    <div className="pf-v5-c-tabs">
      <ul className="pf-v5-c-tabs__list">
        {navItems.map((f, index) => (
          <NavLink
            key={index}
            to={f.path}
            className="pf-v5-c-tabs__item"
            activeClassName="pf-v5-m-current"
          >
            <li key={index} className="pf-v5-c-tabs__item">
              <button className="pf-v5-c-tabs__link">
                <span className="pf-v5-c-tabs__item-text">{f.title}</span>
              </button>
            </li>
          </NavLink>
        ))}
      </ul>
    </div>
  );
};
