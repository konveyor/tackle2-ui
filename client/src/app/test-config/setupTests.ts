import "@testing-library/jest-dom";
import { server } from "@mocks/server";

const mockInitialized = false;

jest.mock("@react-keycloak/web", () => {
  const originalModule = jest.requireActual("@react-keycloak/web");
  return {
    ...originalModule,
    useKeycloak: () => [mockInitialized],
  };
});

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path",
  }),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// Uncomment the following to see request logging in the console:

// server.events.on("request:start", (req) => {
//   console.log(`Handling a request to ${req.url.href}`);
// });

// server.events.on("request:match", (req) => {
//   console.log(`Request to ${req.url.href} was matched with a handler`);
// });

// server.events.on("request:unhandled", (req) => {
//   console.warn(`Request to ${req.url.href} was not handled`);
// });
