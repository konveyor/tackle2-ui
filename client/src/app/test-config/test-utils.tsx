import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
