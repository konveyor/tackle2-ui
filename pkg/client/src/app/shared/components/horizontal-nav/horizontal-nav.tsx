import React from "react";
import { NavLink } from "react-router-dom";

export interface HorizontalNavProps {
  navItems: { title: string; path: string }[];
}

export const HorizontalNav: React.FC<HorizontalNavProps> = ({ navItems }) => {
  return (
    <div className="pf-c-tabs">
      <ul className="pf-c-tabs__list">
        {navItems.map((f, index) => (
          <NavLink
            key={index}
            to={f.path}
            className="pf-c-tabs__item"
            activeClassName="pf-m-current"
          >
            <li key={index} className="pf-c-tabs__item">
              <button className="pf-c-tabs__link">
                <span className="pf-c-tabs__item-text">{f.title}</span>
              </button>
            </li>
          </NavLink>
        ))}
      </ul>
    </div>
  );
};
