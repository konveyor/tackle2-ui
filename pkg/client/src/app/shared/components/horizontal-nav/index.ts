export { HorizontalNav as Welcome } from "./horizontal-nav";
