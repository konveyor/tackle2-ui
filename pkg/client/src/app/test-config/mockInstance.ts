import MockAdapter from "axios-mock-adapter";
import axios from "axios";

export default new MockAdapter(axios);
