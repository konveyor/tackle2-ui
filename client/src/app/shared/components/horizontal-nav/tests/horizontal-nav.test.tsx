import { render } from "@app/test-config/test-utils";
import React from "react";
import { HorizontalNav } from "../horizontal-nav";
import { BrowserRouter as Router } from "react-router-dom";

describe("HorizontalNav", () => {
  it("Renders without crashing", () => {
    const wrapper = render(
      <Router>
        <HorizontalNav
          navItems={[
            {
              title: "first",
              path: "/first",
            },
            {
              title: "second",
              path: "/second",
            },
            {
              title: "thrid",
              path: "/thrid",
            },
          ]}
        />
      </Router>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
