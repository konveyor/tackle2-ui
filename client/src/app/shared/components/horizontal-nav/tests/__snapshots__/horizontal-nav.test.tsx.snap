// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalNav Renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pf-v5-c-tabs"
      >
        <ul
          class="pf-v5-c-tabs__list"
        >
          <a
            class="pf-v5-c-tabs__item"
            href="/first"
          >
            <li
              class="pf-v5-c-tabs__item"
            >
              <button
                class="pf-v5-c-tabs__link"
              >
                <span
                  class="pf-v5-c-tabs__item-text"
                >
                  first
                </span>
              </button>
            </li>
          </a>
          <a
            class="pf-v5-c-tabs__item"
            href="/second"
          >
            <li
              class="pf-v5-c-tabs__item"
            >
              <button
                class="pf-v5-c-tabs__link"
              >
                <span
                  class="pf-v5-c-tabs__item-text"
                >
                  second
                </span>
              </button>
            </li>
          </a>
          <a
            class="pf-v5-c-tabs__item"
            href="/thrid"
          >
            <li
              class="pf-v5-c-tabs__item"
            >
              <button
                class="pf-v5-c-tabs__link"
              >
                <span
                  class="pf-v5-c-tabs__item-text"
                >
                  thrid
                </span>
              </button>
            </li>
          </a>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pf-v5-c-tabs"
    >
      <ul
        class="pf-v5-c-tabs__list"
      >
        <a
          class="pf-v5-c-tabs__item"
          href="/first"
        >
          <li
            class="pf-v5-c-tabs__item"
          >
            <button
              class="pf-v5-c-tabs__link"
            >
              <span
                class="pf-v5-c-tabs__item-text"
              >
                first
              </span>
            </button>
          </li>
        </a>
        <a
          class="pf-v5-c-tabs__item"
          href="/second"
        >
          <li
            class="pf-v5-c-tabs__item"
          >
            <button
              class="pf-v5-c-tabs__link"
            >
              <span
                class="pf-v5-c-tabs__item-text"
              >
                second
              </span>
            </button>
          </li>
        </a>
        <a
          class="pf-v5-c-tabs__item"
          href="/thrid"
        >
          <li
            class="pf-v5-c-tabs__item"
          >
            <button
              class="pf-v5-c-tabs__link"
            >
              <span
                class="pf-v5-c-tabs__item-text"
              >
                thrid
              </span>
            </button>
          </li>
        </a>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
